frontend/app/utils/hcaptcha-utils.ts

```diff
@@ -0,0 +1,45 @@
+import { FormEvent, useEffect, useRef } from 'react';
+
+import { FetcherWithComponents } from '@remix-run/react';
+
+import HCaptcha from '@hcaptcha/react-hcaptcha';
+
+export function useHCaptcha(fetcher: FetcherWithComponents<never>) {
+  const captchaRef = useRef<HCaptcha>(null);
+
+  useEffect(() => {
+    let timeoutId: ReturnType<typeof setTimeout>;
+
+    if (captchaRef.current?.isReady()) {
+      captchaRef.current.execute();
+    } else {
+      timeoutId = setTimeout(() => {
+        captchaRef.current?.execute();
+      }, 500);
+    }
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, []);
+
+  function handleSubmit(event: FormEvent<HTMLFormElement>) {
```

Comment:
I believe `handleSubmit` shouldn't be located here. Perhaps the consumer might want to incorporate additional logic before submission, or handle the `FormData` differently, or use something else than `h-captcha-response` as field name.

Reply:
Good point - fixed.